src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -331,27 +330,32 @@ private async Task ReconnectAsync(CancellationToken cancellationToken)
                 // establishes the first connection, throws if it fails
                 await ConnectFirstAsync(cancellationToken).CfAwait();
 
-                // FIXME reconnect, shall we wait for the first member view event?
-                // else we are just hoping that we haven't switched to a new
-                // cluster and that this connection matches a member?
+                // once the first connection is established, we should use it to subscribe
+                // to the cluster views event, and then we should receive a members view,
+                // which in turn should change the state to Connected - unless something
+                // goes wrong
+                var connected = await _clusterState.WaitForConnectedAsync(cancellationToken).CfAwait();
+
+                if (!connected)
+                {
+                    // we are a background task and cannot throw!
+                    _logger.LogError("Failed to reconnect.");
```

Comment:
Is this intended that we only log error in case of failure, but don't notify caller in any way?

Reply:
Right, that is strange, fixing / adding a notification on success.

Reply:
I thinks only this one is left.